website/docs/language/import/generating-configuration.mdx

```diff
@@ -0,0 +1,137 @@
+│ 
+│ Generating configuration during import is currently experimental, and the generated configuration format may change in future versions.
+╵
+
+──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
+
+Terraform has generated configuration and written it to generated.tf. Please review the configuration and edit it as necessary before adding it to version control.
+```
+
+### 3. Review Generated Configuration
+
+In our example, we specified that Terraform should generate configuration in a file named `generated.tf`. The below code demonstrates what the `generated.tf` file might look like. 
+
+```hcl
+resource aws_iot_thing "bar" {
+  name = "foo"
+}
+```
+
+You can review the code Terraform generated and add or remove resource arguments, variables, and references to other resources as needed for your configuration. 
+
+### 4. Apply
+Running `terraform apply` create and provisions your infrastructure. Ensure you add your generated code alongside your Terraform code to your version control system.
+
+```
+$ terraform apply
+
+aws_iot_thing.bar: Preparing import... [id=foo]
+aws_iot_thing.bar: Refreshing state... [id=foo]
+
+Terraform will perform the following actions:
+
+  # aws_iot_thing.bar will be imported
+    resource "aws_iot_thing" "bar" {
+        arn               = "arn:aws:iot:eu-west-1:1234567890:thing/foo"
+        attributes        = {}
+        default_client_id = "foo"
+        id                = "foo"
+        name              = "foo"
+        version           = 1
+    }
+
+Plan: 1 to import, 0 to add, 0 to change, 0 to destroy.
+aws_iot_thing.bar: Importing... [id=foo]
+aws_iot_thing.bar: Import complete [id=foo]
+
+Apply complete! Resources: 1 imported, 0 added, 0 changed, 0 destroyed.
+
+```
+
+Make sure that your generated code, along with your other Terraform code, is added to your version control system.
+
+## Limitations
+
+Terraform generates configuration for importable resources during a plan by requesting values for resource attributes from the provider. For certain resources with complex schemas, it may not be possible for Terraform to construct a valid configuration from these values.
+
+Terraform will throw an error like the one below if it does not receive values for resource attributes while generating configuration.
+
+```
+$ terraform plan -generate-config-out=generated.tf
+╷
+│ Error: Conflicting configuration arguments
+│ 
+│   with aws_instance.ubuntu,
+│   on g.tf line 20, in resource "aws_instance" "ubuntu":
+│   20:   ipv6_address_count                   = 0
+│ 
+│ "ipv6_address_count": conflicts with ipv6_addresses
+╵
+```
+
+In the example above, Terraform still generates configuration and writes it to `generated.tf`. This error stems from a conflict between the `ipv6_address_count` and `ipv6_addresses` arguments. You could fix the error by removing one of these two arguments.
```

Comment:
(Adding this in case you didn't see the comment, but you can ignore if you did) Are there any other errors folks might run into? The error here feels pretty clear, but I'm trying to think if we should highlight any weird edge cases?

Reply:
Thanks. This is the main weird one for now that we haven't mentioned elsewhere. We may expand this section later (hopefully not too much!).